packages/docusaurus/bin/beforeCli.mjs

```diff
@@ -105,16 +106,31 @@ export default async function beforeCli() {
       .map((p) => p.concat('@latest'))
       .join(' ');
 
+    const getYarnVersion = async () => {
+      const fallbackYarnVersion = 1;
+      const yarnVersionResult = shell.exec('yarn --version', {silent: true});
+      if (yarnVersionResult?.code === 0) {
+        const yarnVersionOutput = yarnVersionResult.stdout?.trim();
+        const majorVersion = parseInt(
+          yarnVersionOutput?.split('.')[0] ?? '',
+          10,
+        );
+        if (typeof majorVersion === 'number' && !Number.isNaN(majorVersion)) {
+          return majorVersion;
```

Comment:
Doesn't this return Yarn version even if the current project use npm? We should only suggest Yarn if there is actually a `yarn.lock` file.

Reply:
You're right. This can be a problem if yarn is installed globally, but the project uses npm.

In the if statement on line 118, I probably need to add a conditional to see if the `yarn.lock` file exists. Also, I probably don't need the `fallbackYarnVersion` variable with the if statement on lines 122-124.

Reply:
Improved execution speed when using npm by making the `getYarnVersion()` method check for the existence of `yarn.lock` before running.